en_US/advanced/retained.md

```diff
@@ -8,18 +8,32 @@ When a client establishes a subscription, and there are retained messages with m
 
 The `emqx_retainer` plugin is enabled by default, and the configuration path of the plugin is `etc/plugins/emqx_retainer.conf`.
 
-| Configuration item       | Type  | Optional value      | Default value | Description                                               |
-| ------------------------------ | -------- | ------------------------ | ------ | ------------------------------------------------------------ |
-| retainer.storage_type          | enum     | `ram`, `disc`, `disc_only` | ram |ram: only stored in memory; <br /> disc: stored in memory and hard disk; <br /> disc_only: only stored in hard disk|
-| retainer.max_retained_messages | integer  | \>= 0                    | 1      | The maximum number of retained messages, and 0 means no limit. After the number of retained messages exceeds the maximum limit, you can replace the existing retained messages, but cannot store retained messages for new topics. |
-| retainer.max_payload_size      | bytesize |                          | 1MB    | Retain the maximum Payload value of the message. After the Payload value exceeds the maximum value, the EMQX broker will treat the retained reserved message as a normal message. |
-| retainer.expiry_interval       | duration |                          | ０     | The expiration time of retaining message, and 0 means never expire. If the message expiration interval is set in the PUBLISH packet, the message expiration interval in the PUBLISH packet shall prevail. |
+| Configuration item                         | Type     | Possible values             | Default value | Description                                                                                                                                                                                                                        |
+| ------------------------------------------ | -------- | -------------------------- | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| retainer.enable                            | boolean  | `false`, `true`            | `true`        | Enables retainer plugin                                                                                                                                                                                                            |
+| retainer.msg_clear_interval                | time     |                            | `0s`          | Periodic interval for cleaning up expired messages, and 0 means never expire.                                                                                                                                                     |
```

Comment:
How does this interact with the expiration intervals from PUBLISH packets?

Reply:
Hmm. I'll need to check this

Reply:
Timeout which occurs first, removes the message from retainment